drivers/pwm/pwm_led_esp32.c

```diff
@@ -279,6 +279,10 @@ static int pwm_led_esp32_set_cycles(const struct device *dev, uint32_t channel_i
 		return -EINVAL;
 	}
 
+	if (flags & PWM_POLARITY_INVERTED) {
+		pulse_cycles = period_cycles - pulse_cycles;
```

Comment:
Would you mind checking period_cycles is greater than or equal to "pulse_cycles" before proceed with the subtraction?